server/src/main/java/org/elasticsearch/action/update/UpdateHelper.java

```diff
@@ -181,7 +181,7 @@ static String calculateRouting(GetResult getResult, @Nullable IndexRequest updat
     Result prepareUpdateIndexRequest(ShardId shardId, UpdateRequest request, GetResult getResult, boolean detectNoop) {
         final IndexRequest currentRequest = request.doc();
         final String routing = calculateRouting(getResult, currentRequest);
-        final DocumentSizeObserver documentSizeObserver = documentParsingProvider.newDocumentSizeObserver();
+        final DocumentSizeObserver documentSizeObserver = documentParsingProvider.newDocumentSizeObserver(request);
         final Tuple<XContentType, Map<String, Object>> sourceAndContent = XContentHelper.convertToMap(getResult.internalSourceRef(), true);
```

Comment:
I wish we could somehow get access to that field that we create for RA-S here .. with that we could keep the original size instead potentially affecting it with the script

Reply:
we agreed that update by scripts should be metered for RA-S but not RA-I.